lib/modules/datasource/index.ts

```diff
@@ -218,6 +220,9 @@ function resolveRegistryUrls(
   } else if (is.nonEmptyArray(defaultRegistryUrls)) {
     resolvedUrls = [...defaultRegistryUrls];
     resolvedUrls.concat(additionalRegistryUrls ?? []);
+  } else if (is.function_(datasource.defaultRegistryUrls)) {
```

Comment:
this case needs a test

Reply:
I choose to keep the `getRegistryUrl` function for that.

Reply:
please add a test witjh the dummy datasource, should be very easy